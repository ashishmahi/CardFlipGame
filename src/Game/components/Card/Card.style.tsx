import styled from 'styled-components/native';

export const CardWrapper = styled.View`
  margin: 10px;
  border-width: 4px;
  border-style: solid;
  border-color: white;
  width: 27%;
  height: 140px;
  justify-content: center;
  align-items: center;
`;
export const CardContent = styled.Text`
  font-size: 30px;
  color: white;
`;
