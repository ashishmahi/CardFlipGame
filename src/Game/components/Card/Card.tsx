import React from 'react';
import {CardContent, CardWrapper} from './Card.style';

type CardProps = {
  id: number;
  isOpen: boolean;
};
const Card: React.FC<CardProps> = ({isOpen, id}) => {
  return (
    <CardWrapper
      style={{
        backgroundColor: isOpen ? 'white' : 'deepskyblue',
      }}>
      <CardContent
        style={{
          color: isOpen ? 'black' : 'white',
        }}>
        {isOpen ? id : '?'}
      </CardContent>
    </CardWrapper>
  );
};

export default Card;
