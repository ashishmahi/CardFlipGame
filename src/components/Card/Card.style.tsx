import styled from 'styled-components/native';

export const CardWrapper = styled.Pressable`
  margin: 10px;
  border-width: 4px;
  border-style: solid;
  border-color: white;
  width: 27%;
  height: 140px;
  justify-content: center;
  align-items: center;
  background-color: ${(props: {isOpen: boolean}) =>
    props.isOpen ? 'white' : 'deepskyblue'};
`;
export const CardContent = styled.Text`
  font-size: 30px;
  color: ${(props: {isOpen: boolean}) => (props.isOpen ? 'black' : 'white')};
`;
