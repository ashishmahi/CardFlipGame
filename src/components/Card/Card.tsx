import React from 'react';
import {CardContent, CardWrapper} from './Card.style';

export type CardModal = {
  id: number;
  number: number;
};
type CardProps = {
  card: CardModal;
  isOpen: boolean;
  onCardPress: (id: number) => void;
};
const Card: React.FC<CardProps> = ({isOpen, card, onCardPress}) => {
  return (
    <CardWrapper
      onPress={onCardPress}
      testID={`card-${card.id}`}
      isOpen={isOpen}>
      <CardContent testID={`card-content-${card.id}`} isOpen={isOpen}>
        {isOpen ? card.number : '?'}
      </CardContent>
    </CardWrapper>
  );
};

export default Card;
