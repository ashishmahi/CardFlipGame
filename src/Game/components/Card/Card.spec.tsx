import React from 'react';
import {render} from '@testing-library/react-native';
import Card from './Card';
describe('Card Test', () => {
  it('should render closed card', () => {
    expect(render(<Card id={1} isOpen={false} />)).toMatchSnapshot();
  });

  it('should render open card', () => {
    expect(render(<Card id={1} isOpen />)).toMatchSnapshot();
  });
});
