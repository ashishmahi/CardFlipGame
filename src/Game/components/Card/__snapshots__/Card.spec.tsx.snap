// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test should render closed card 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderStyle": "solid",
        "borderWidth": 4,
        "height": 140,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "width": "27%",
      },
      Object {
        "backgroundColor": "deepskyblue",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    ?
  </Text>
</View>
`;

exports[`Card Test should render open card 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderStyle": "solid",
        "borderWidth": 4,
        "height": 140,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "width": "27%",
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;
